overlord/snapstate/handlers.go

```diff
@@ -836,16 +838,21 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	snapst.Active = false
 
-	// do the final unlink
-	linkCtx := backend.LinkContext{
-		FirstInstall: false,
-		// This task is only used for unlinking a snap during refreshes so we
-		// can safely hard-code this condition here.
-		RunInhibitHint: runinhibit.HintInhibitedForRefresh,
-	}
-	err = m.backend.UnlinkSnap(oldInfo, linkCtx, NewTaskProgressAdapterLocked(t))
-	if err != nil {
-		return err
+	if oldInfo.Type() != snap.TypeSnapd {
+		// snapd current symlink on the refresh path can only replaced
+		// by a symlink to a new revision of the snapd snap
```

Comment:
this comment being inside the if was initially confusing to me and made me think that the `!= snap.TypeSnapd` was meant to be a `== snap.TypeSnapd`, but I see what the intention was, can you move this comment outside the if so it is more clear that this if branch is only taken in the case that we aren't the unlinking snapd snap?

i.e.

```suggestion
	// snapd current symlink on the refresh path can only replaced
	// by a symlink to a new revision of the snapd snap, so only do
	// the actual unlink if we're not working on the snapd snap
	if oldInfo.Type() != snap.TypeSnapd {
```